### PR TITLE
update .net SDK to OperationalInsights package-2020-10

### PR DESCRIPTION
**Pull request by dabenhamMic**
swagger pr: https://github.com/Azure/azure-rest-api-specs/pull/13782

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [v] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running generate.ps1. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version,

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by dabenhamMic** [comment]
manually created the pr using this command: (my swagger pr was not yet approved - so using the generate.ps1 script generated the "old" sdk code)

autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/operationalinsights/resource-manager/readme.md --csharp --version=v2 --reflect-api-versions --tag=package-2020-10 --csharp-sdks-folder=C:\One\SDK_repos\azure-sdk-for-net\sdk

**Comment by dabenhamMic** [comment]
@JonathanCrd could you please advise me what shall I do in order to promote this pr?

**Comment by JonathanCrd** [comment]
Hi @dabenhamMic , please re run the command so the metadata text file is up to date now that your Swagger PR is approved.

PRs for this repo should be created once the Swagger PR has been merged.

**Comment by dabenhamMic** [comment]
@JonathanCrd  , the swagger is up to date, I'm unable to invoke the "/azp run net - mgmt - ci" command

File: sdk/operationalinsights/Microsoft.Azure.Management.OperationalInsights/src/Generated/Models/Workspace.cs [hunk 1]
```diff
@@ -51,27 +51,39 @@ public Workspace()
...
+        public Workspace(string location, string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), string provisioningState = default(string), string customerId = default(string), WorkspaceSku sku = default(WorkspaceSku), int? retentionInDays = default(int?), WorkspaceCapping workspaceCapping = default(WorkspaceCapping), string createdDate = default(string), string modifiedDate = default(string), string publicNetworkAccessForIngestion = default(string), string publicNetworkAccessForQuery = default(string), bool? forceCmkForQuery = default(bool?), IList<PrivateLinkScopedResource> privateLinkScopedResources = default(IList<PrivateLinkScopedResource>), IDictionary<string, object> features = default(IDictionary<string, object>), string eTag = default(string))
```
Comment (JonathanCrd) [inline sdk/operationalinsights/Microsoft.Azure.Management.OperationalInsights/src/Generated/Models/Workspace.cs:72]: There is a breaking change here, new parameters should be added at the end.
Please rearrange the parameters in the swagger file.

File: sdk/operationalinsights/Microsoft.Azure.Management.OperationalInsights/src/Generated/Models/WorkspacePatch.cs [hunk 2]
```diff
@@ -49,27 +49,39 @@ public WorkspacePatch()
...
+        public WorkspacePatch(string id = default(string), string name = default(string), string type = default(string), string etag = default(string), string provisioningState = default(string), string customerId = default(string), WorkspaceSku sku = default(WorkspaceSku), int? retentionInDays = default(int?), WorkspaceCapping workspaceCapping = default(WorkspaceCapping), string createdDate = default(string), string modifiedDate = default(string), string publicNetworkAccessForIngestion = default(string), string publicNetworkAccessForQuery = default(string), bool? forceCmkForQuery = default(bool?), IList<PrivateLinkScopedResource> privateLinkScopedResources = default(IList<PrivateLinkScopedResource>), IDictionary<string, object> features = default(IDictionary<string, object>), IDictionary<string, string> tags = default(IDictionary<string, string>))
```
Comment (JonathanCrd) [inline sdk/operationalinsights/Microsoft.Azure.Management.OperationalInsights/src/Generated/Models/WorkspacePatch.cs:70]: Same here

**Review by JonathanCrd — CHANGES_REQUESTED** [review]
Found 2 breaking changes regarding new parameters.
Please fix those, everything else looks good to me.

File: sdk/operationalinsights/Microsoft.Azure.Management.OperationalInsights/src/Generated/Models/Workspace.cs (hunk 1 above)
Comment (dabenhamMic) [reply sdk/operationalinsights/Microsoft.Azure.Management.OperationalInsights/src/Generated/Models/Workspace.cs]: moved new parameters to the end

File: sdk/operationalinsights/Microsoft.Azure.Management.OperationalInsights/src/Generated/Models/WorkspacePatch.cs (hunk 2 above)
Comment (dabenhamMic) [reply sdk/operationalinsights/Microsoft.Azure.Management.OperationalInsights/src/Generated/Models/WorkspacePatch.cs]: moved new parameters to the end

**Comment by dabenhamMic** [comment]
@JonathanCrd , I finished updating your comments, I'm unable to invoke the "/azp run net - mgmt - ci" command.

**Comment by dabenhamMic** [comment]
@JonathanCrd , I re ran the generate.ps1 script, please invoke the "/azp run net - mgmt - ci" command and merge if possible.

**Review by JonathanCrd — APPROVED** [review]

**Pull request merged**